podcasts/AutoplayHelper.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+class AutoplayHelper {
+    static let shared = AutoplayHelper()
+
+    private let userDefaults: UserDefaults
+    private let userDefaultsKey = "playlist"
+
+    var lastPlaylist: EpisodesDataManager.Playlist? {
+        if let rawDictionary = userDefaults.dictionary(forKey: userDefaultsKey),
+           let dictData = try? JSONSerialization.data(withJSONObject: rawDictionary) {
+            return try? JSONDecoder().decode(EpisodesDataManager.Playlist.self, from: dictData)
+        }
+
+        return nil
+    }
+
+    init(userDefaults: UserDefaults = UserDefaults.standard) {
+        self.userDefaults = userDefaults
+    }
+
+    func playedAt(playlist: EpisodesDataManager.Playlist?) {
+        #if !os(watchOS)
+        save(selectedPlaylist: playlist)
+        #endif
+    }
+
+    private func save(selectedPlaylist playlist: EpisodesDataManager.Playlist?) {
+        guard let playlist else {
+            userDefaults.removeObject(forKey: userDefaultsKey)
+            return
+        }
+
+        if let data = try? JSONEncoder().encode(playlist),
+           let dict = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) {
+            userDefaults.set(dict, forKey: userDefaultsKey)
+        }
```

Comment:
Should we add some debug FileLog's if we're not able to save or restore the playlist? It could help debugging issues in the future.

Reply:
Done in d1c1df32cd2be054b82f755a17a6d2a28193f970